packages/office-ui-fabric-react/src/components/Check/Check.styles.ts

```diff
@@ -139,3 +139,17 @@ export const getStyles = (props: ICheckStyleProps): ICheckStyles => {
     checkHost: [{}]
   };
 };
+
+/**
+ * Style which should be applied to the parent element of the check, so that the check's opacity
+ * is set when the parent is hovered/focused. This replaces `ICheckStyles.checkHost` because
+ * calculating the check styles in the parent component causes an unacceptable perf penalty.
+ * (Global class names must be enabled for this to work.)
+ */
+export const CHECK_HOST_HOVER_STYLE: IRawStyle = {
+  selectors: {
+    [`:hover, :hover .${GlobalClassNames.root}, :focus, :focus .${GlobalClassNames.root}`]: {
+      opacity: 1
+    }
```

Comment:
Hmm, I thought one of the things using `getGlobalClassNames` allows consumers to do is affect default global naming usage. Is that prevented by having this const refer to `GlobalClassNames` directly instead of using `getGlobalClassNames` output?

Reply:
`getGlobalClassNames` is for disabling global class names, in which case this style will entirely break. I'm not sure of a good way to help set up check host styling if global class names are disabled.